src/main.ts

```diff
@@ -227,6 +237,26 @@ function toClosureJS(
       let {output} = tsickle.processES5(
           fileName, moduleId, jsFiles.get(fileName)!, cliSupport.pathToModuleName);
       jsFiles.set(fileName, output);
+    } else {
+      const tscSourceMapJson: any = jsFiles.get(fileName);
+      const tscSourceMap = new SourceMapConsumer(tscSourceMapJson);
+      const tscSourceMapGenerator = SourceMapGenerator.fromSourceMap(tscSourceMap);
+
+      for (const sourceFileName of (new SourceMapConsumer(tscSourceMapJson) as any).sources) {
+        const tsickleSourceMapGenerator = tsickleSourceMaps.get(sourceFileName);
+        if (!tsickleSourceMapGenerator) {
+          console.log('couldn\'t find a souce map for: ' + sourceFileName);
+          return null;
+        }
+        const tsickleRawSourceMap = tsickleSourceMapGenerator.toJSON();
+        tsickleRawSourceMap.sources[0] = sourceFileName;
```

Comment:
That was super confusing, I ended up changing rewriter.ts to make it unnecessary.